include/fplus/benchmark_session.hpp

```diff
@@ -0,0 +1,359 @@
+// Copyright 2015, Tobias Hermann and the FunctionalPlus contributors.
+// https://github.com/Dobiasd/FunctionalPlus
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+
+#include <vector>
+#include <mutex>
+#include <fplus/fplus.hpp>
+
+namespace fplus
+{
+using FunctionName = std::string;
+struct benchmark_function_report
+{
+    size_t nb_calls;
+    ExecutionTime total_time;
+    ExecutionTime average_time;
+    ExecutionTime deviation;
+};
+
+
+namespace internal
+{
+    std::string show_benchmark_function_report(const std::map<FunctionName, benchmark_function_report> & reports);
+}
+
+
+// benchmark_session stores timings during a benchmark session
+// and is able to emit a report at the end
+class benchmark_session
+{
+public:
+    benchmark_session() : functions_times_mutex_(), functions_times_() {};
+
+    // report() shall return a string with a summary of the session
+    // Example below:
+    // Function              |Nb calls|Total time|Av. time|Deviation|
+    // ----------------------+--------+----------+--------+---------+
+    // convert_charset_string|    4000|   4.942ms| 1.236ns|  1.390ns|
+    // split_lines           |    1000|   4.528ms| 4.528ns|  1.896ns|
+    inline std::string report() const 
+    {
+        const auto reports = report_list();
+        return fplus::internal::show_benchmark_function_report(reports);
+    }
+
+    std::map<FunctionName, benchmark_function_report> report_list() const
+    {
+        std::lock_guard<std::mutex> lock(functions_times_mutex_);
+        std::map<FunctionName, benchmark_function_report> report;
+        for (const auto & one_function_time : functions_times_)
+            report[one_function_time.first] = make_bench_report(one_function_time.second);
```

Comment:
Please use

```cpp
for (...)
{
   loop-body
}
```

Instead of

```cpp
for (...)
   loop-body
```
I know the first version is not strictly/objective better, but matches the code style.

Reply:
ok